public/app/features/templating/formatRegistry.ts

```diff
@@ -13,15 +14,36 @@ export interface FormatOptions {
 }
 
 export interface FormatRegistryItem extends RegistryItem {
+  canHandle(variable: VariableModel): boolean;
   formatter(options: FormatOptions, variable: VariableModel): string;
 }
 
+export enum FormatRegistryID {
+  lucene = 'lucene',
+  raw = 'raw',
+  regex = 'regex',
+  pipe = 'pipe',
+  distributed = 'distributed',
+  csv = 'csv',
+  html = 'html',
+  json = 'json',
+  percentencode = 'percentencode',
+  singlequote = 'singlequote',
+  doublequote = 'doublequote',
+  sqlstring = 'sqlstring',
+  date = 'date',
+  glob = 'glob',
+  text = 'text',
+  queryparam = 'queryparam',
```

Comment:
minor nit: think typescript enums are generally PascalCase.

Reply:
Thank you, I'll fix it!

Reply:
Fixed!